shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Ppup.hs

```diff
@@ -38,12 +38,15 @@ import Shelley.Spec.Ledger.LedgerState (PPUPState (..), pvCanFollow)
 import Shelley.Spec.Ledger.PParams
 import Shelley.Spec.Ledger.Serialization (decodeRecordSum)
 import Shelley.Spec.Ledger.Slot
+import qualified Cardano.Ledger.Core as Core
 
 data PPUP era
 
 data PPUPEnv era
   = PPUPEnv SlotNo (PParams era) (GenDelegs (Crypto era))
 
+type instance Core.UPEnv era = PPUPEnv era
```

Comment:
This instance would make `Core.UPEnv era` a `PPUPEnv` for every era. I suspect that's not what you want.

I imagine you want concrete instances for `ShelleyEra` (and others) setting it to `PPUPEnv`.

```
type instance Core.UPEnv (ShelleyEra c) = PPUPEnv (ShelleyEra c)
```

Then a future era would be able to set it to a different type without conflicting.

This otherwise seems like the right direction.

Reply:
That makes sense. Thanks!